python/cuspatial/cuspatial/geometry/geoarrowbuffers.py

```diff
@@ -0,0 +1,640 @@
+            MultiPoint((2, 0), (2, 1), (2, 2)),
+        ])
+
+    LineString geometry is more complicated than multipoints because the
+    format allows for the use of LineStrings and MultiLineStrings in the same
+    buffer, via the mlines key::
+
+        buffers = GeoArrowBuffers({
+            "lines_xy":
+                [0, 0, 0, 1, 0, 2, 1, 0, 1, 1, 1, 2, 2, 0, 2, 1, 2, 2, 3, 0,
+                 3, 1, 3, 2, 4, 0, 4, 1, 4, 2],
+            "lines_offsets":
+                [0, 6, 12, 18, 24, 30],
+            "mlines":
+                [1, 3]
+        })
+
+    Which encodes a GeoPandas Series::
+
+        series = geopandas.Series([
+            LineString((0, 0), (0, 1), (0, 2)),
+            MultiLineString([(1, 0), (1, 1), (1, 2)],
+                            [(2, 0), (2, 1), (2, 2)],
+            )
+            LineString((3, 0), (3, 1), (3, 2)),
+            LineString((4, 0), (4, 1), (4, 2)),
+        ])
+
+    Polygon geometry includes `mpolygons` for MultiPolygons similar to the
+    LineString geometry. Polygons are encoded using the same format as
+    Shapefiles, with left-wound external rings and right-wound internal rings.
+    An exact example of `GeoArrowBuffers` to `geopandas.Series` is left to the
+    reader as an exercise. Convert any GeoPandas `Series` or `DataFrame` with
+    `cuspatial.from_geopandas(geopandas_object)`.
+
+    Notes
+    -----
+    Legacy cuspatial algorithms depend on separated x and y columns. Access
+    them with the `.x` and `.y` properties.
+
+    Examples
+    --------
+    GeoArrowBuffers accept a dict as argument. Valid keys are in the bullet
+    list above. Valid values are any datatype that implements numpy's
+    `__array_interface__`. Any or all of the four basic geometry types is
+    supported as argument::
+
+        buffers = GeoArrowBuffers({
+            "points_xy":
+                [0, 0, 0, 1, 0, 2, 1, 0, 1, 1, 1, 2, 2, 0, 2, 1, 2, 2],
+            "multipoints_xy":
+                [0, 0, 0, 1, 0, 2, 1, 0, 1, 1, 1, 2, 2, 0, 2, 1, 2, 2],
+            "multipoints_offsets":
+                [0, 6, 12, 18]
+            "lines_xy":
+                [0, 0, 0, 1, 0, 2, 1, 0, 1, 1, 1, 2, 2, 0, 2, 1, 2, 2, 3, 0,
+                 3, 1, 3, 2, 4, 0, 4, 1, 4, 2],
+            "lines_offsets":
+                [0, 6, 12, 18, 24, 30],
+            "mlines":
+                [1, 3]
+            "polygons_xy":
+                [0, 0, 0, 1, 0, 2, 1, 0, 1, 1, 1, 2, 2, 0, 2, 1, 2, 2, 3, 0,
+                 3, 1, 3, 2, 4, 0, 4, 1, 4, 2],
+            "polygons_polygons": [0, 1, 2],
+            "polygons_rings": [0, 1, 2],
+            "mpolygons": [1, 3],
+        })
+
+    or another GeoArrowBuffers::
+
+        buffers2 = GeoArrowBuffers(buffers)
+    """
+
+    def __init__(self, data: Union[dict, T], data_locale: object = cudf):
+        """
+        Parameters
+        ----------
+        data : A dict or a GeoArrowBuffers object.
+        """
```

Comment:
I think we generally would want to move this docstring into the class docstring?

Reply:
as you wish